### PR TITLE
Mccalluc/main test

### PR DESCRIPTION
**Pull request by mccalluc**
@sgratzl : ok to merge if tests pass?

### CODE REVIEW

**Comment by sgratzl** [comment]
according to https://github.com/phovea/phovea.github.io/pull/34/files -> the actual reponsible persons are @thinkh  and @aaljuhani for everything related to the core. Isn't that the reason why we have the list?

File: tests/index.test.ts [hunk 1]
```diff
@@ -1,34 +1,137 @@
...
+});
+
+describe('bounds', () => {
+  /* TODO: This seems odd. For instance, there is already an x and y provided by the DOM,
+  but we give a different meaning to these.  */
+  it('not a DOM object', () => expect(bounds(null)).toEqual({ x: 0, y: 0, w: 0, h: 0 }));
+  // TODO: DOM object
+});
+
+it('version', () => expect(version).toEqual('__VERSION__') );
```
Comment (sgratzl) [inline tests/index.test.ts:29]: strange __VERSION__ should be replaced by the actual version during build time

File: tests/index.test.ts [hunk 2]
```diff
@@ -1,34 +1,137 @@
...
+// TODO: (1) Doesn't seem right. (2) Does anything actually need it?
+
+describe('uniqueId', () => {
+  it('first time', () => expect(uniqueId()).toEqual(0));
+  it('second time', () => expect(uniqueId()).toEqual(1));
+});
+
+describe('uniqueString', () => {
+  it('first time', () => expect(uniqueString()).toEqual('_default2'));
```
Comment (sgratzl) [inline tests/index.test.ts:38]: what happens if I swap the order of `uniqueId` and `uniqueString`?

File: tests/index.test.ts [hunk 3]
```diff
@@ -1,34 +1,137 @@
...
+    });
   });
 });
+
+describe('randomId', () => {
+  it('default', () => expect(randomId()).toMatch(/^[0-9a-z]{8}$/));
+  it('short', () => expect(randomId(1)).toMatch(/^[0-9a-z]$/));
+});
+
+// TODO: What is this for?
```
Comment (sgratzl) [inline tests/index.test.ts:60]: interacts with `uniqueId` you are flagging that the given id is already used

**Review by sgratzl — APPROVED** [review]
well done

File: tests/index.test.ts (hunk 2 above)
Comment (mccalluc) [reply tests/index.test.ts:38]: Order dependence in the tests is admittedly not a good thing... If there is a way to reset the internal counter between tests, that would be good?

**Pull request merged**